code/modules/mob/living/silicon/robot/robot_defense.dm

```diff
@@ -82,12 +82,8 @@
 	. = ..()
 	if(. & EMP_PROTECT_SELF)
 		return
-	switch(severity)
-		if(1)
-			Stun(160)
-		if(2)
-			Stun(60)
-
+	if(severity) //catch to prevent divide by zero
+		Stun(30/severity)
```

Comment:
3 seconds is very short, especially since EMPs are rarer and should be more powerful than flashes. This should have some other effects like disabling modules, slowing down or blinding or something.

Reply:
It does blind for seven seconds, EMPs call the flash proc in addition to their stun.

I can make it disable all modules too though, that seems like a cool added effect. Cyborg can just re-equip them, but it does add some additional delay before they can react.

Reply:
Okay:
* Doubled damage (10/20 -> 20/40)
* Disables Modules
* Blinds like flash (it already did this, just pointing it out again) 
* It will slow down via the damage it deals.